tests/autorag_research/nodes/retrieval/test_bm25.py

```diff
@@ -0,0 +1,235 @@
+            BM25Module()
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_init_with_index_path(self, mock_lucene_class):
+        """Should initialize with custom index path."""
+        mock_searcher = MagicMock()
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+
+        mock_lucene_class.assert_called_once_with("/path/to/index")
+        mock_searcher.set_language.assert_called_once_with("en")
+        mock_searcher.set_bm25.assert_called_once_with(k1=0.9, b=0.4)
+        assert module.index_name is None
+        assert module.index_path is not None
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_init_with_prebuilt_index(self, mock_lucene_class):
+        """Should initialize with pre-built index."""
+        mock_searcher = MagicMock()
+        mock_lucene_class.from_prebuilt_index.return_value = mock_searcher
+
+        module = BM25Module(index_name="msmarco-passage")
+
+        mock_lucene_class.from_prebuilt_index.assert_called_once_with("msmarco-passage")
+        mock_searcher.set_language.assert_called_once_with("en")
+        mock_searcher.set_bm25.assert_called_once_with(k1=0.9, b=0.4)
+        assert module.index_name == "msmarco-passage"
+        assert module.index_path is None
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_init_with_custom_bm25_params(self, mock_lucene_class):
+        """Should set custom BM25 parameters."""
+        mock_searcher = MagicMock()
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index", k1=1.2, b=0.75)
+
+        mock_searcher.set_bm25.assert_called_once_with(k1=1.2, b=0.75)
+        assert module.k1 == 1.2
+        assert module.b == 0.75
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_init_with_custom_language(self, mock_lucene_class):
+        """Should set custom language."""
+        mock_searcher = MagicMock()
+        mock_lucene_class.return_value = mock_searcher
+
+        BM25Module(index_path="/path/to/index", language="ko")
+
+        mock_searcher.set_language.assert_called_once_with("ko")
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_init_prefers_index_name_over_path(self, mock_lucene_class):
+        """When both index_name and index_path are provided, index_name takes precedence."""
+        mock_searcher = MagicMock()
+        mock_lucene_class.from_prebuilt_index.return_value = mock_searcher
+
+        module = BM25Module(index_name="msmarco-passage", index_path="/path/to/index")
+
+        mock_lucene_class.from_prebuilt_index.assert_called_once_with("msmarco-passage")
+        mock_lucene_class.assert_not_called()
+        assert module.index_name == "msmarco-passage"
+        assert module.index_path is None
```

Comment:
I don't like these test codes. These test codes are for checking if it is well initialized, but let's be honest. Why the hell do we want to test the "initialization phase"?? 
It looks like totally AI bullshit to me. (overengineered test code) 
I already made AI instructions for testing, but it looks like your opus did not look at the file.